examples/k8s-4-action/charmcraft.yaml

```
@@ -0,0 +1,61 @@
+type: charm
+bases:
+  - build-on:
+    - name: ubuntu
+      channel: "22.04"
+    run-on:
+    - name: ubuntu
+      channel: "22.04"
+
+name: demo-api-charm
+title: |
+  demo-fastapi-k8s
+description: |
+  This is a demo charm built on top of a small Python FastAPI server.
+  This charm can be integrated with the PostgreSQL charm and COS Lite bundle (Canonical Observability Stack).
+summary: |
+  FastAPI Demo charm for Kubernetes
+
+assumes:
+  - juju >= 3.1
+  - k8s-api
+
+config:
+  options:
+    server-port:
+      default: 8000
+      description: Default port on which FastAPI is available
+      type: int
+
+requires:
+  database:
+    interface: postgresql_client
+    limit: 1
```

Feel free to say that we should leave this for a follow-up PR: we now strongly recommend that the `optional` field is always included for all the provides/requires relations, rather than relying on the default.
```suggestion
  database:
    interface: postgresql_client
    limit: 1
    optional: false
```